src/Generator.cpp

```diff
@@ -1047,27 +1050,49 @@ void execute_generator(const ExecuteGeneratorArgs &args_in) {
     if (!args.generator_name.empty()) {
         const std::string base_path = args.output_dir + "/" + args.file_base_name;
         debug(1) << "Generator " << args.generator_name << " has base_path " << base_path << "\n";
+
+        // Factory function for creating a generator instance for a given function name and target
+        auto generator_factory = [&](const std::string &function_name, const Target &target) -> AbstractGeneratorPtr {
+            // Must re-create each time since each instance will have a different Target.
+            auto gen = args.create_generator(args.generator_name, GeneratorContext(target));
+            for (const auto &kv : args.generator_params) {
+                if (kv.first == "target") {
+                    continue;
+                }
+                gen->set_generatorparam_value(kv.first, kv.second);
+            }
+            return gen;
+        };
+
         if (args.output_types.count(OutputFileType::cpp_stub)) {
-            // When generating cpp_stub, we ignore all generator args passed in, and supply a fake Target.
+            // When generating cpp_stub we ignore all generator args passed in, and supply a fake Target.
             // (CompilerLogger is never enabled for cpp_stub, for now anyway.)
             const Target fake_target = Target();
             auto gen = args.create_generator(args.generator_name, GeneratorContext(fake_target));
             auto output_files = compute_output_files(fake_target, base_path, args.output_types);
-            gen->emit_cpp_stub(output_files[OutputFileType::cpp_stub]);
+            if (args.output_types.count(OutputFileType::cpp_stub)) {
+                gen->emit_cpp_stub(output_files[OutputFileType::cpp_stub]);
+            }
+        }
+
+        if (args.output_types.count(OutputFileType::hlpipe)) {
+            // When serializing a halide pipeline, target is required (since the schedule may be target dependent)
+            // If multiple targets are specified, add the target name as a suffix to the filename
+            const bool use_target_suffix = (args.targets.size() > 1);
+            for (size_t i = 0; i < args.targets.size(); ++i) {
+                const Target &target = args.targets[i];
+                const std::string hlpipe_path = use_target_suffix ? (base_path + "-" + target.to_string()) : base_path;
+                auto output_files = compute_output_files(target, hlpipe_path, args.output_types);
+                auto gen = generator_factory(args.function_name, target);
+                gen->emit_hlpipe(output_files[OutputFileType::hlpipe]);
+            }
         }
 
-        // Don't bother with this if we're just emitting a cpp_stub.
+        // Don't bother with this if we're just emitting a cpp_stub
```

Comment:
Why did you remove the period? This is a complete sentence, so it should have proper punctuation.

Reply:
Will fix.